### ISSUE TITLE
[Bug] emacs does not come with an executable in the PATH on macOS by default

### ISSUE THREAD

**Issue by jdtsmith** [alexluigit/dirvish#168]
### Thank you for the bug report

- [X] I am using the latest version of `dirvish` related packages.
- [X] I checked [CUSTOMIZING|EXTENSIONS](https://github.com/alexluigit/dirvish#resources).
- [ ] You may also try reproduce the issue using clean environment and minimal configurations with
the command `emacs -Q`.

### Bug description

```
Debugger entered--Lisp error: (file-missing "Searching for program" "No such file or directory" "emacs")
  make-process(:name "dir-data" :connection-type nil :buffer "dir-data-gHbKC1" :command ("emacs" "-Q" "-batch" "--eval" "(message \"%s\" (with-temp-buffer (let ((hash (make-hash-table)) (bk (ignore-errors (vc-responsible-backend \"~/\")))) (dolist (file (directory-files \"~/\"...") :sentinel dirvish-dir-data-proc-s)
 ```

### Steps to reproduce

Invoke `dirvish-layout-toggle` while in dirvish sidebar.  Not sure why an external emacs process is needed, but assuming there is an "emacs" command is problematic for MacOS, where Emacs.app is the typical launch point. 

### Expected behavior

Formerly this was dirvish-fullscreen, but it seems that was removed.  

### OS

MacOS

### Emacs Version

28

### Emacs Configurations

n/a

### Error callstack

```emacs-lisp
see above
```

**Comment by alexluigit** [comment]
> Invoke `dirvish-layout-toggle` while in dirvish sidebar.

Does it also happen for regular dirvish buffer?

> but assuming there is an "emacs" command is problematic for MacOS, where Emacs.app is the typical launch point.

Thanks for the reminder. Dirvish call "emacs" command in multiple places (actually every time you enter a directory), if this is a problem, why wouldn't emacs raise an error earlier? Anyway, I'll try to reproduce it in a macOS machine. Thank you for the report!

**Comment by alexluigit** [comment]
I believe this issue has been resolved by f34761c.
